ui_framework/src/components/code_editor/__snapshots__/code_editor.test.js.snap

```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`KuiCodeEditor is rendered 1`] = `"<div class=\\"kuiCodeEditorWrapper\\"><div class=\\"kuiCodeEditorKeyboardHint\\" id=\\"42\\" tabindex=\\"0\\" role=\\"button\\"><p class=\\"kuiText kuiVerticalRhythmSmall\\">Press Enter to start editing.</p><p class=\\"kuiText kuiVerticalRhythmSmall\\">When you’re done, press Escape to stop editing.</p></div><div id=\\"brace-editor\\" style=\\"width: 500px; height: 500px;\\" class=\\" ace_editor ace-tm testClass1 testClass2\\"><textarea class=\\"ace_text-input\\" wrap=\\"off\\" autocorrect=\\"off\\" autocapitalize=\\"off\\" spellcheck=\\"false\\" style=\\"opacity: 0;\\" tabindex=\\"-1\\"></textarea><div class=\\"ace_gutter\\"><div class=\\"ace_layer ace_gutter-layer ace_folding-enabled\\"></div><div class=\\"ace_gutter-active-line\\"></div></div><div class=\\"ace_scroller\\"><div class=\\"ace_content\\"><div class=\\"ace_layer ace_print-margin-layer\\"><div class=\\"ace_print-margin\\" style=\\"left: 4px; visibility: visible;\\"></div></div><div class=\\"ace_layer ace_marker-layer\\"></div><div class=\\"ace_layer ace_text-layer\\" style=\\"padding: 0px 4px;\\"></div><div class=\\"ace_layer ace_marker-layer\\"></div><div class=\\"ace_layer ace_cursor-layer ace_hidden-cursors\\"><div class=\\"ace_cursor\\"></div></div></div></div><div class=\\"ace_scrollbar ace_scrollbar-v\\" style=\\"display: none; width: 20px;\\"><div class=\\"ace_scrollbar-inner\\" style=\\"width: 20px;\\"></div></div><div class=\\"ace_scrollbar ace_scrollbar-h\\" style=\\"display: none; height: 20px;\\"><div class=\\"ace_scrollbar-inner\\" style=\\"height: 20px;\\"></div></div><div style=\\"height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;\\"><div style=\\"height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;\\"></div><div style=\\"height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;\\">XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX</div></div></div></div>"`;
```

Comment:
This looks really bad, and likely "impossible" to review if there are changes to the component.

Reply:
And what's up with `XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX`?

Reply:
This was done as @cjcenizal suggested to render out the whole DOM of ACE, and not just the more or less empty wrapper element. I still have mixed feelings about rendering library code in general and compare it, but I would leave the level that we test this snapshot up to your discussion :-)

Reply:
Oof, I didn't realize it wasn't going to be formatted!  It's not so bad when it's formatted, but then there are all the escape double-quotes, too... bah...

```html
<div class=\\ "kuiCodeEditorWrapper\\">
    <div class=\\ "kuiCodeEditorKeyboardHint\\" id=\\ "42\\" tabindex=\\ "0\\" role=\\ "button\\">
        <p class=\\ "kuiText kuiVerticalRhythmSmall\\">Press Enter to start editing.</p>
        <p class=\\ "kuiText kuiVerticalRhythmSmall\\">When you’re done, press Escape to stop editing.</p>
    </div>
    <div id=\\ "brace-editor\\" style=\\ "width: 500px; height: 500px;\\" class=\\ " ace_editor ace-tm testClass1 testClass2\\">
        <textarea class=\\ "ace_text-input\\" wrap=\\ "off\\" autocorrect=\\ "off\\" autocapitalize=\\ "off\\" spellcheck=\\ "false\\" style=\\ "opacity: 0;\\" tabindex=\\ "-1\\"></textarea>
        <div class=\\ "ace_gutter\\">
            <div class=\\ "ace_layer ace_gutter-layer ace_folding-enabled\\"></div>
            <div class=\\ "ace_gutter-active-line\\"></div>
        </div>
        <div class=\\ "ace_scroller\\">
            <div class=\\ "ace_content\\">
                <div class=\\ "ace_layer ace_print-margin-layer\\">
                    <div class=\\ "ace_print-margin\\" style=\\ "left: 4px; visibility: visible;\\"></div>
                </div>
                <div class=\\ "ace_layer ace_marker-layer\\"></div>
                <div class=\\ "ace_layer ace_text-layer\\" style=\\ "padding: 0px 4px;\\"></div>
                <div class=\\ "ace_layer ace_marker-layer\\"></div>
                <div class=\\ "ace_layer ace_cursor-layer ace_hidden-cursors\\">
                    <div class=\\ "ace_cursor\\"></div>
                </div>
            </div>
        </div>
        <div class=\\ "ace_scrollbar ace_scrollbar-v\\" style=\\ "display: none; width: 20px;\\">
            <div class=\\ "ace_scrollbar-inner\\" style=\\ "width: 20px;\\"></div>
        </div>
        <div class=\\ "ace_scrollbar ace_scrollbar-h\\" style=\\ "display: none; height: 20px;\\">
            <div class=\\ "ace_scrollbar-inner\\" style=\\ "height: 20px;\\"></div>
        </div>
        <div style=\\ "height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;\\">
            <div style=\\ "height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;\\"></div>
            <div style=\\ "height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;\\">XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX</div>
        </div>
    </div>
</div>
```

Reply:
@kjbekkelund @timroes Which of these three options do you think makes the most sense?

### 1) Use `debug()`

```javascript
expect(mount(component).debug()).toMatchSnapshot();
```

Yields:

```jsx
<KuiCodeEditor aria-label="aria-label" className="testClass1 testClass2" data-test-subj="test subject string">
  <div className="kuiCodeEditorWrapper" style={{...}}>
    <div className="kuiCodeEditorKeyboardHint" id={42} tabIndex="0" role="button" onClick={[Function]} onKeyDown={[Function]}>
      <p className="kuiText kuiVerticalRhythmSmall">
        Press Enter to start editing.
      </p>
      <p className="kuiText kuiVerticalRhythmSmall">
        When you’re done, press Escape to stop editing.
      </p>
    </div>
    <ReactAce aria-label="aria-label" className="testClass1 testClass2" data-test-subj="test subject string" onBlur={[Function]} name="brace-editor" focus={false} mode="" theme="" height="500px" width="500px" value="" fontSize={12} showGutter={true} onChange={{...}} onPaste={{...}} onLoad={{...}} onScroll={{...}} minLines={{...}} maxLines={{...}} readOnly={false} highlightActiveLine={true} showPrintMargin={true} tabSize={4} cursorStart={1} editorProps={{...}} setOptions={{...}} wrapEnabled={false} enableBasicAutocompletion={false} enableLiveAutocompletion={false}>
      <div id="brace-editor" style={{...}} />
    </ReactAce>
  </div>
</KuiCodeEditor>
```

### 2) Use prettified `html()`

```javascript
// Note that this requires the `html` dependency, which the UI Framework already uses.
import html from 'html';
expect(html.prettyPrint(mount(component).html(), {
  indent_size: 2,
  unformatted: [], // Expand all tags, including spans
})).toMatchSnapshot();
```

Yields:

```html
<div class="kuiCodeEditorWrapper">
  <div class="kuiCodeEditorKeyboardHint" id="42" tabindex="0" role="button">
    <p class="kuiText kuiVerticalRhythmSmall">Press Enter to start editing.</p>
    <p class="kuiText kuiVerticalRhythmSmall">When you’re done, press Escape to stop editing.</p>
  </div>
  <div id="brace-editor" style="width: 500px; height: 500px;" class=" ace_editor ace-tm testClass1 testClass2">
    <textarea class="ace_text-input" wrap="off" autocorrect="off" autocapitalize="off"
    spellcheck="false" style="opacity: 0;" tabindex="-1"></textarea>
    <div class="ace_gutter">
      <div class="ace_layer ace_gutter-layer ace_folding-enabled"></div>
      <div class="ace_gutter-active-line"></div>
    </div>
    <div class="ace_scroller">
      <div class="ace_content">
        <div class="ace_layer ace_print-margin-layer">
          <div class="ace_print-margin" style="left: 4px; visibility: visible;"></div>
        </div>
        <div class="ace_layer ace_marker-layer"></div>
        <div class="ace_layer ace_text-layer" style="padding: 0px 4px;"></div>
        <div class="ace_layer ace_marker-layer"></div>
        <div class="ace_layer ace_cursor-layer ace_hidden-cursors">
          <div class="ace_cursor"></div>
        </div>
      </div>
    </div>
    <div class="ace_scrollbar ace_scrollbar-v" style="display: none; width: 20px;">
      <div class="ace_scrollbar-inner" style="width: 20px;"></div>
    </div>
    <div class="ace_scrollbar ace_scrollbar-h" style="display: none; height: 20px;">
      <div class="ace_scrollbar-inner" style="height: 20px;"></div>
    </div>
    <div style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;">
      <div style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"></div>
      <div style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;">XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX</div>
    </div>
  </div>
</div>
```

### 3) Use `render()`, AKA you're over-thinking it CJ

```jsx
expect(render(component)).toMatchSnapshot();
```

Yields:

```html
<div
  class="kuiCodeEditorWrapper"
>
  <div
    class="kuiCodeEditorKeyboardHint"
    id="42"
    role="button"
    tabindex="0"
  >
    <p
      class="kuiText kuiVerticalRhythmSmall"
    >
      Press Enter to start editing.
    </p>
    <p
      class="kuiText kuiVerticalRhythmSmall"
    >
      When you’re done, press Escape to stop editing.
    </p>
  </div>
  <div
    id="brace-editor"
    style="width:500px;height:500px;"
  />
</div>
```